pkg/querier/queryrange/query_range.go

```diff
@@ -234,17 +235,16 @@ func (prometheusCodec) DecodeResponse(ctx context.Context, r *http.Response, _ R
 		body, _ := ioutil.ReadAll(r.Body)
 		return nil, httpgrpc.Errorf(r.StatusCode, string(body))
 	}
-
-	sp, _ := opentracing.StartSpanFromContext(ctx, "ParseQueryRangeResponse")
-	defer sp.Finish()
+	log, _ := spanlogger.New(ctx, "ParseQueryRangeResponse")
```

Comment:
New span here. Should replace the context, no?

Reply:
I'm basing this off how we used the ot library `sp, _ := opentracing.StartSpanFromContext(ctx, "ParseQueryRangeResponse")`, where we ignored the returned context. I don't know the historical reason behind doing it that way, so if you think it should change here with the introduction of spanlogger that's fine by me.

Reply:
I think the reason is that `ctx` is not used below within this function. I believe replacing it is more future proof to avoid mistakes (we should do it once we'll start using `ctx` below and it's easy to forget to do it).